server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -458,7 +458,8 @@ public static boolean canReserveCompaction(TabletMetadata tablet, CompactionKind
               tablet.getExtent(), userRequestedCompactions);
           return false;
         } else if (tablet.getSelectedFiles() != null
-            && !Collections.disjoint(jobFiles, tablet.getSelectedFiles().getFiles())) {
+            && (tablet.getSelectedFiles().getCompletedJobs() > 0
```

Comment:
Could simplify this else if to just

```java
} else if (!Collections.disjoint(jobFiles, getFilesResevedBySelection(tablet, steadyTime, ctx))) {
```

with this function 

```java
  private static Set<StoredTabletFile> getFilesResevedBySelection(TabletMetadata tabletMetadata, SteadyTime steadyTime, ServerContext ctx) {
    if(tabletMetadata.getSelectedFiles() == null) {
      return Set.of();
    }

    if(tabletMetadata.getSelectedFiles().getCompletedJobs() > 0) {
      return tabletMetadata.getSelectedFiles().getFiles();
    }

    long selectedExpirationDuration = ctx.getTableConfiguration(tabletMetadata.getTableId()).getTimeInMillis(Property.TABLE_COMPACTION_SELECTION_EXPIRATION);


    if(steadyTime.minus(tabletMetadata.getSelectedFiles().getSelectedTime()).toMillis() > selectedExpirationDuration) {
      return tabletMetadata.getSelectedFiles().getFiles();
    }

    return Set.of();
  }
```

this uses the behavior of the empty set w/ Collections.disjoint to make the "else if" simpler. It also includes checks for expiration.

Reply:
This has been implemented in the latest commit, dafdc87afc but I just wanted to note that the logic above is backwards. Time time check should be: 

```java
if(steadyTime.minus(tabletMetadata.getSelectedFiles().getSelectedTime()).toMillis() < selectedExpirationDuration) {
   return tabletMetadata.getSelectedFiles().getFiles();
}
```